### PR TITLE
Refactor: replace cudamalloc and cudafree with cuda_mem_wrapper

### PR DESCRIPTION
**Pull request by dzzz2001**
### What's changed?
- modified the GPU memory allocation strategy, which overall reduced the GPU memory consumption for grid integration calculation.
- unified variable names in the GPU code of gint.
- optimize some functions.

### CODE REVIEW

File: source/module_hamilt_lcao/module_gint/gint_force_gpu.cu [hunk 1]
```diff
@@ -56,231 +24,304 @@ namespace GintKernel
...
+    Cuda_Mem_Wrapper<double> force(3 * nat, num_streams, true);
+    Cuda_Mem_Wrapper<double> stress(6, num_streams, true);
+    Cuda_Mem_Wrapper<int> iat_per_z(max_atom_per_z, num_streams, true);
+
+    Cuda_Mem_Wrapper<double> dm_matrix(lgd * lgd, 1, true);
+    for (int iat1 = 0; iat1 < ucell.nat; iat1++)
+    {
+        for (int iat2 = 0; iat2 < ucell.nat; iat2++)
+        {
+            int it1 = ucell.iat2it[iat1];
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force_gpu.cu]: add const if possible: const int it1

Comment (dzzz2001) [reply source/module_hamilt_lcao/module_gint/gint_force_gpu.cu]: fixed

File: source/module_hamilt_lcao/module_gint/gint_vl_gpu.cu [hunk 2]
```diff
@@ -36,311 +36,208 @@ namespace GintKernel
...
+
+    Cuda_Mem_Wrapper<int> gemm_m(max_atompair_per_z, num_streams, true);
+    Cuda_Mem_Wrapper<int> gemm_n(max_atompair_per_z, num_streams, true);
+    Cuda_Mem_Wrapper<int> gemm_k(max_atompair_per_z, num_streams, true);
+    Cuda_Mem_Wrapper<int> gemm_lda(max_atompair_per_z, num_streams, true);
+    Cuda_Mem_Wrapper<int> gemm_ldb(max_atompair_per_z, num_streams, true);
+    Cuda_Mem_Wrapper<int> gemm_ldc(max_atompair_per_z, num_streams, true);
+    Cuda_Mem_Wrapper<double*> gemm_A(max_atompair_per_z, num_streams, true);
+    Cuda_Mem_Wrapper<double*> gemm_B(max_atompair_per_z, num_streams, true);
+    Cuda_Mem_Wrapper<double*> gemm_C(max_atompair_per_z, num_streams, true);
```
Comment (A-006) [inline source/module_hamilt_lcao/module_gint/gint_vl_gpu.cu:111]: gemm_{m,n,k,lda,ldb,ldc,A,B,C},psi_input_{double,int} can use a structure to store it, making it convenient for the three calculations.

Comment (mohanchen) [reply source/module_hamilt_lcao/module_gint/gint_vl_gpu.cu:111]: OK, maybe we can fix it next time

**Review by mohanchen — APPROVED** [review]

**Pull request merged**